python map and filter builtin func additions (#603)
Codecademy/docs: content/python/concepts/built-in-functions/terms/map/map.md

---
Title: 'map()'
Description: 'Returns an iterator that takes a function and applies it to every item in an iterable.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Functions'
  - 'Methods'
  - 'Strings'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
---

> The `map()` built-in function takes a function and applies it to every item in an iterable. The output of `map()` is a map object, in the examples below `list()` is applied to the map expression to return a new list.

Reviewer: ```suggestion
The `map()` built-in function accepts a function and applies it to every item in an iterable. It outputs a map object. In the examples below, the `list()` function is applied to the map expression to return a new list.
```

Reviewer: If you choose to use `list()` in the `filter()` entry, can you put a similar explanation like this one mentioned here?

Reviewer: In addition, can we move this line to right before the examples? So for this entry, on line 29?

Author: Sorry, not clear on what you're suggesting. Do want to move the new (last) sentence in your edit from line 17 to 29?

Reviewer: @caupolicandiaz The edits you made work as well! Thanks for incorporating this suggestion 🙏

---
Title: 'map()'

> Description: 'A built-in function that takes a function and applies it to every item in an iterable.'

Reviewer 2: ```suggestion
Description: 'Takes a function and applies it to every item in an iterable.'
```

Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Functions'
  - 'Methods'
  - 'Strings'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
---

> 'A built-in function that takes a function and applies it to every item in an iterable. The output of `map()` is a map object, in the examples below `list()` is applied to the map expression to return a new list.'

Reviewer 3: ```suggestion
The `map()` built-in function takes a function and applies it to every item in an iterable. The output of `map()` is a map object, in the examples below `list()` is applied to the map expression to return a new list.
```

## Syntax

> ```py

Reviewer 2: ```suggestion
```pseudo
```

> ```psuedo

Reviewer 2: ```suggestion
```pseudo
```

map(func_name, iterable)
```

> ## Example 1

Reviewer 2: ```suggestion
## Example
```

Use `map()` to return the length of each string in the following list:

> ```codebyte/python

Reviewer 2: ```suggestion
```py
```

> ## Example 1

Reviewer 2: ```suggestion
## Example
```

some_strings = ['learn', 'how', 'to', 'code', 'online']

print(list(map(len, some_strings))

# Output: [5, 3, 2, 4, 6]
```

> ## Example 2

Reviewer 2: ```suggestion
## Codebyte Example
```

> Use `map()` to return the length of each string in the following list:

Reviewer 2: ```suggestion
The following example uses `map()` to return the length of each string in the list `some_strings`:
```

> Use `map()` to change all the list items to integers:

Reviewer 2: ```suggestion
The `map()` function can be used to transform all the list items to integers:
```

Reviewer 2: Any chance we can rephrase without the imperitive?

```codebyte/python
a_list = [1.6, 5.4,12.0]

> print(list(map(lambda x: int(x), a_list))

Reviewer 2: Same as mentioned in my earlier comment in `filter.md`, I think it would be better to delineate this line into a helper function followed by the `print()` statement, like so:

```py
def make_int(x):
  return int(x)

print(list(map(make_int, a_list)))
```

There was also a missing closing parenthesis `)` at the end of the `print()` statement.

Author: I think having it's valuable to represent different inputs and the previous example demonstrates a built-in func so I used a lambda in this case

Reviewer 2: That's super fair. I actually just modified my suggestion above on line # 39 for you to take another look at. 😄

> print(list(map(len, some_strings)))
> 

Reviewer 2: ```suggestion
print(list(map(len, some_strings)))
```

> 
> # Output: [1, 5, 12]

Reviewer 2: ```suggestion
```

> ## Example 2

Reviewer 2: ```suggestion
## Codebyte Example
```

---
Title: 'map()'
Description: 'Returns an iterator that takes a function and applies it to every item in an iterable.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Functions'
  - 'Methods'
  - 'Strings'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
---

The `map()` built-in function accepts a function and applies it to every item in an iterable. It outputs a map object.

## Syntax

```pseudo
map(func_name, iterable)
```

> The `func_name` is the function to be applied to the iterable and can be a lambda function or the name of any defined function. The `iterable`, such as a list, contains the items the function will act on.

Reviewer: ```suggestion
- The `func_name` is the function to be applied to the iterable and can be a lambda function or the name of any defined function.
- The `iterable`, such as a list, contains the items the function will act on.
```

## Example 1

Use `map()` to return the length of each string in the following list:

```py
some_strings = ['learn', 'how', 'to', 'code', 'online']

> print(list(map(len, some_strings))

Reviewer 2: ```suggestion
print(list(map(len, some_strings)))
```

This seems to be missing a closing parenthesis `)`.

# Output: [5, 3, 2, 4, 6]
```

## Example 2

The `map()` function can be used to transform all the list items to integers:

```codebyte/python
a_list = [1.6, 5.4,12.0]

> print(list(map(lambda x: int(x), a_list)))
> 

Reviewer 2: ```suggestion
print(list(map(lambda x: int(x), a_list)))
```

Author: Sorry, I don't see the suggested change here. Please confirm

Reviewer 2: Delete line # 47, please.
